### ISSUE TITLE
HTTP url storage provider should do retires & manage parallel downloads

### ISSUE THREAD

**Issue by enricorotundo** [filecoin-project/bacalhau#411]
Luke:

> should we do retries? maybe make a follow-up issue for it.
> 
> also: do we currently download all URLs in parallel, because they are separate storage driver instances? this is probably ok for now, but if user provides millions of URLs this might overload the system and we'd want to put them in some sort of capped parallelism queue

See https://github.com/filecoin-project/bacalhau/pull/398#discussion_r929727492
